### PR TITLE
Fix #29472 - nullptr while trying to find a NPC in the overmap

### LINKED ISSUE #29472: Segfault on map open or after few moves
**Describe the bug**  
Game just crash when I open map (press 'm'), or make few moves.

```CRASH LOG FILE: config/crash.log
VERSION: 0.D-1679-gdd4532a
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:
    @0x5583A5[cataclysm-tiles.exe+0x1583A5]
    @0x558F52[cataclysm-tiles.exe+0x158F52]
    SMPEG_error+0x4B034@0xE3F030[cataclysm-tiles.exe+0xA3F030]
    _C_specific_handler+0x9C@0x77BB85A8[ntdll.dll+0x185A8]
    RtlDecodePointer+0xBD@0x77BC9D0D[ntdll.dll+0x29D0D]
    RtlUnwindEx+0xBBF@0x77BB91AF[ntdll.dll+0x191AF]
    KiUserExceptionDispatcher+0x2E@0x77BF1278[ntdll.dll+0x51278]
    @0xAF2221[cataclysm-tiles.exe+0x6F2221]
    @0xAF7CCD[cataclysm-tiles.exe+0x6F7CCD]
    @0xAFF2FB[cataclysm-tiles.exe+0x6FF2FB]
    @0x69C468[cataclysm-tiles.exe+0x29C468]
    @0x6800C7[cataclysm-tiles.exe+0x2800C7]
    IMG_LoadWEBP_RW+0x4E16C9@0x1412569[cataclysm-tiles.exe+0x1012569]
    @0x4013ED[cataclysm-tiles.exe+0x13ED]
    @0x4014FB[cataclysm-tiles.exe+0x14FB]
    BaseThreadInitThunk+0xD@0x77A9652D[kernel32.dll+0x1652D]
    RtlUserThreadStart+0x21@0x77BCC521[ntdll.dll+0x2C521]
```

**To Reproduce**  
I provide my savegame and config folder, so it's enough to load that savegame and press 'm'.

**Versions and configuration(please complete the following information):**
 - OS: Windows 7 x64
 - Game Version: 0.D-1679-gdd4532a
 - Graphics version: Tiles
 - Mods loaded: ```[
  "dda",
  "no_npc_food",
  "novitamins",
  "More_Survival_Tools",
  "mutant_npcs",
  "national_guard_camp",
  "more_locations",
  "Urban_Development",
  "boats",
  "blazemod",
  "Tanks",
  "no_olg_guns",
  "no_religious_Texts",
  "no_reviving_zombies",
  "realguns",
  "growable-pots",
  "modular_turrets",
  "Salvaged_Robots",
  "FujiStruct"
]```

**Additional context**  
Link to a savegame: https://yadi.sk/d/gJvEHkncdmGOMg

### PR DESCRIPTION
**Pull request by neitsa**
#### Summary

```SUMMARY: Bugfixes "Fix #29472 - nullptr while trying to find a NCP in the overmap"```

#### Purpose of change

Fix #29472 - nullptr while trying to find a NCP in the overmap

There were multiple missing checks on `find_npc` method which might return `nullptr` if the NCP can't be found in the overmap.

#### Describe the solution

* [X] Add the missing check for the crash issue.
* [X] Check all other locations where the function is used.

Add a simple check for `nullptr`. Add some debug msgs in missions to better diagnose the missing NPC problem.

#### Describe alternatives you've considered

There are probably no simpler alternatives.

#### Additional context

N/A.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -3423,6 +3426,9 @@ void game::disp_NPC_epilogues()
     // TODO: This search needs to be expanded to all NPCs
     for( auto elem : follower_ids ) {
         std::shared_ptr<npc> npc_to_get = overmap_buffer.find_npc( elem );
+        if( ! npc_to_get ) {
```
Comment (Night-Pryanik) [inline src/game.cpp]:
```suggestion
        if( !npc_to_get ) {
```

File: src/game.cpp [hunk 2]
```diff
@@ -4683,6 +4689,9 @@ void game::overmap_npc_move()
     // for now just processing NPC followers on travelling missions
     for( auto &elem : get_follower_list() ) {
         std::shared_ptr<npc> npc_to_get = overmap_buffer.find_npc( elem );
+        if( ! npc_to_get ) {
```
Comment (Night-Pryanik) [inline src/game.cpp]:
```suggestion
        if( !npc_to_get ) {
```

File: src/game.cpp [hunk 3]
```diff
@@ -3188,7 +3191,7 @@ void game::debug()
 #if defined(TILES)
             const point offset {
                 POSX - u.posx() + u.view_offset.x,
-                     POSY - u.posy() + u.view_offset.y
+                POSY - u.posy() + u.view_offset.y
```
Comment (ifreund) [inline src/game.cpp]: Astyle on the CI servers doesn't like this for some unknown reason.  See #29465
```suggestion
                    POSY - u.posy() + u.view_offset.y
```

Comment (neitsa) [reply src/game.cpp]: I'm using Astyle 3.1 at the moment. I guess it's related to #29465.

**Pull request merged**